### PR TITLE
feat(overlay): set overlay size

### PR DESCRIPTION
**Pull request by kara**
This PR adds two properties to the overlay state to manually configure the size of the overlay.

r: @jelbourn

### CODE REVIEW

File: src/lib/core/overlay/overlay-state.ts [hunk 1]
```diff
@@ -12,10 +12,16 @@ export class OverlayState {
...
   hasBackdrop: boolean = false;
 
+  /** Custom class to add to the backdrop **/
   backdropClass: string = 'md-overlay-dark-backdrop';
 
+  /** The width of the overlay panel **/
+  width: number;
+
+  /** The height of the overlay panel **/
+  height: number;
```
Comment (jelbourn) [inline src/lib/core/overlay/overlay-state.ts]: We should support percent with this as well. I'd make the type `number|string`, assume numbers are `px` and strings are whole units.

File: src/lib/core/overlay/overlay.spec.ts [hunk 2]
```diff
@@ -98,6 +98,30 @@ describe('Overlay', () => {
...
+      overlay.create(state).attach(componentPortal);
+      const pane = (<HTMLElement>overlayContainerElement.children[0]);
+      expect(pane.style.width).toEqual('500px');
+    });
+
+    it('should apply the height set in the config', () => {
+      state.height = 500;
+
+      overlay.create(state).attach(componentPortal);
+      const pane = (<HTMLElement>overlayContainerElement.children[0]);
```
Comment (jelbourn) [inline src/lib/core/overlay/overlay.spec.ts]: I want to start preferring the `as` cast syntax.

File: src/lib/core/overlay/overlay-ref.ts [hunk 3]
```diff
@@ -58,6 +59,17 @@ export class OverlayRef implements PortalHost {
     }
   }
 
+  /** Updates the size of the overlay based on the overlay config. */
+  updateSize() {
+    if (this._state.width) {
```
Comment (jelbourn) [inline src/lib/core/overlay/overlay-ref.ts]: It should _probably_ work for zero as well.

File: src/lib/core/overlay/overlay-ref.ts [hunk 4]
```diff
@@ -98,3 +110,7 @@ export class OverlayRef implements PortalHost {
     }
   }
 }
+
+export function coerceToCSSFormat(value: number | string) {
```
Comment (jelbourn) [inline src/lib/core/overlay/overlay-ref.ts]: `formatCssUnit` ?

Also it doesn't need to be exported just yet

**Comment by kara** [comment]
@jelbourn Comments addressed!

**Pull request merged**